emission/analysis/intake/segmentation/section_segmentation.py

```diff
@@ -99,14 +133,42 @@ def segment_trip_into_sections(user_id, trip_entry, trip_source):
     # set it properly.
     ts = esta.TimeSeries.get_time_series(user_id)
 
-    get_loc_for_ts = lambda time: ecwl.Location(ts.get_entry_at_ts("background/filtered_location", "data.ts", time)["data"])
+    # ------------------------------------------------------------------
+    # Define a function to use preloaded filtered locations instead of querying on each point.
+    # Logs indicate whether the preloaded lookup or the fallback method is used.
+    # Looks like preloaded lookup is used for all points, so the fallback method is not used -- well according to tests that is
+    # ------------------------------------------------------------------
+    def get_loc_for_ts(time):
+        if time in filtered_loc_lookup:
+            logging.info("Using preloaded filtered location for time: %s" % time)
+            return ecwl.Location(filtered_loc_lookup[time]["data"])
+        else:
+            logging.info("Using fallback get_entry_at_ts for time: %s" % time)
+            entry = ts.get_entry_at_ts("background/filtered_location", "data.ts", time)
+            return ecwl.Location(entry["data"])
```

Comment:
We just preloaded all the filtered_location entries. What situation would we need this fallback?